contracts/contracts/lib/SubnetIDHelper.sol

```diff
@@ -130,13 +128,13 @@ library SubnetIDHelper {
     /// down in the path from the subnet id given as argument.
     /// subnet2 needs to be a prefix of the subnet1.
     /// If subnet1 is /a/b/c/d and subnet2 is /a/b, then the returned ID should be /a/b/c.
-    /// @dev Revert will be triggered if subnet2 is an invalid input.
+    /// @dev Returns an empty SubnetID if subnet2 is not a prefix of subnet1 or if the roots are different.
     function down(SubnetID calldata subnet1, SubnetID calldata subnet2) public pure returns (SubnetID memory) {
         if (subnet1.root != subnet2.root) {
-            revert DifferentRootNetwork();
+            return SubnetID({root: 0, route: new address[](0)});
```

Comment:
is there a reason why it's not reverted? Feels like it's an error that should be handled instead of introducing `root = 0` as invalid subnet. Probably in some use cases `root: 0, route: []` is a valid use case.

Reply:
The problem here is that if I revert, I lose control elsewhere. The method is also used in`validateCrossMessage`, where reverting is not desirable. Additionally, `commonParent` method here returns the empty subnet ID instead of reverting. But I understand your point—if this were deployed on Ethereum, the root could be zero. Would returning a boolean (indicating whether it was found or not) help?